libs/muddle/include/muddle/muddle_interface.hpp

```diff
@@ -0,0 +1,243 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "muddle/address.hpp"
+#include "network/uri.hpp"
+
+#include <vector>
+#include <string>
+#include <cstdint>
+#include <memory>
+#include <unordered_set>
+#include <unordered_map>
+
+namespace fetch {
+namespace crypto {
+  class Prover;
+  class Identity;
+}
+
+namespace network {
+  class NetworkManager;
+}
+
+namespace muddle {
+
+class MuddleEndpoint;
+class NetworkId;
+
+class MuddleInterface
+{
+public:
+  enum class Confidence
+  {
+    DEFAULT,
+    WHITELIST,
+    BLACKLIST
+  };
+
+  using Peers         = std::unordered_set<std::string>;
+  using Ports         = std::vector<uint16_t>;
+  using Addresses     = std::unordered_set<Address>;
+  using ConfidenceMap = std::unordered_map<Address, Confidence>;
+  using AddressHints  = std::unordered_map<Address, network::Uri>;
+
+  // Construction / Destruction
+  MuddleInterface() = default;
+  virtual ~MuddleInterface() = default;
+
+  /// @name Muddle Setup
+  /// @{
+
+  /**
+   * Start the muddle instance connecting to the initial set of peers and listing on the specified
+   * set of ports
+   *
+   * @param peers The initial set of peers that muddle should connect to
+   * @param ports The set of ports to listen on. Zero signals a random port
+   * @return true if successful, otherwise false
+   */
+  virtual bool Start(Peers const &peers, Ports const &ports) = 0;
+
+  /**
+   * Stop the muddle instance, this will cause all the connected to close
+   */
+  virtual void Stop() = 0;
+
+  /**
+   * Get the endpoint interface for this muddle instance/
+   *
+   * @return The endpoint pointer
+   */
+  virtual MuddleEndpoint &GetEndpoint() = 0;
+
+  /// @}
+
+  /// @name Muddle Status
+  /// @{
+
+  /**
+   * Get the associated network for this muddle instance
+   *
+   * @return The current network
+   */
+  virtual NetworkId const &GetNetwork() const = 0;
+
+  /**
+   * Get the address of this muddle node
+   *
+   * @return The address of the node
+   */
+  virtual Address const &GetAddress() const = 0;
```

Comment:
It seems that `GetAdress` and `GetLocalAddress` does the same thing. I'd suggest to streamline the names.

Reply:
done